runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SdkComponents.java

```diff
@@ -54,9 +54,11 @@
 
   private final Set<String> reservedIds = new HashSet<>();
 
+  private String defaultEnvironmentId;
+
   /** Create a new {@link SdkComponents} with no components. */
   public static SdkComponents create() {
-    return new SdkComponents("");
+    return new SdkComponents(null, "");
```

Comment:
I would pass a default instance of `SdkComponents` instead of null, e.g. `RunnerApi.Components.getDefaultInstance()`.